File: DynamicKey/AgoraDynamicKey/nodejs/sample/sample_builder.js
var SimpleTokenBuilder = require('../src/SimpleTokenBuilder').SimpleTokenBuilder;
var Role = require('../src/SimpleTokenBuilder').Role;
var Priviledges = require('../src/AccessToken').priviledges;
var appID  = "970CA35de60c44645bbae8a215061b33";
var appCertificate     = "5CFd2fd1755d40ecb72977518be15d3b";
var channel = "7d72365eb983485397e3e3f9d460bdda";
var uid = 2882341273;
var expireTimestamp = 0;

var builder = new SimpleTokenBuilder(appID, appCertificate, channel, uid);

builder.initPrivileges(Role.kRoleAttendee);
builder.setPrivilege(Priviledges.kJoinChannel, expireTimestamp);
builder.setPrivilege(Priviledges.kPublishAudioStream, expireTimestamp);
builder.setPrivilege(Priviledges.kPublishVideoStream, expireTimestamp);
builder.setPrivilege(Priviledges.kPublishDataStream, expireTimestamp);

var token = builder.buildToken();
console.log(token);

File: DynamicKey/AgoraDynamicKey/python/sample/sample_builder.py
#! /usr/bin/python
# ! -*- coding: utf-8 -*-

import sys
import unittest
import os
import time
from random import randint

sys.path.append(os.path.join(os.path.dirname(__file__), '../src'))
import SimpleTokenBuilder
import AccessToken

appID = "970CA35de60c44645bbae8a215061b33"
appCertificate = "5CFd2fd1755d40ecb72977518be15d3b"
channelName = "7d72365eb983485397e3e3f9d460bdda"
uid = 2882341273
expireTimestamp = 0


def main():
    builder = SimpleTokenBuilder.SimpleTokenBuilder(appID, appCertificate, channelName, uid)
    builder.initPrivileges(SimpleTokenBuilder.Role_Attendee)
    builder.setPrivilege(AccessToken.kJoinChannel, expireTimestamp)
    builder.setPrivilege(AccessToken.kPublishAudioStream, expireTimestamp)
    builder.setPrivilege(AccessToken.kPublishVideoStream, expireTimestamp)
    builder.setPrivilege(AccessToken.kPublishDataStream, expireTimestamp)

    result = builder.buildToken()
    print result


if __name__ == "__main__":
    main()
